lib/deprecate.js

```diff
@@ -0,0 +1,12 @@
+module.exports = deprecate
+
+const loggedMessages = {}
+
+function deprecate (message) {
+  if (loggedMessages[message]) {
+    return
+  }
+
+  console.warn(`DEPRECATED (@octokit/rest): ${message}`)
+  loggedMessages[message] = 1
+}
```

Comment:
this stores logged messages globally in memory. If `@octokit/rest` is instantiated multiple times then a message logged by a previous instance will not be logged again, which I think is right behaviour here. And there is not a high risk of a memory leak because there is a finite number of unique deprecation messages